apps/wallet/src/dapp-interface/WalletStandardInterface.ts

```diff
@@ -0,0 +1,185 @@
+
+type WalletEventsMap = {
+    [E in WalletEventNames]: Parameters<WalletEvents[E]>[0];
+};
+
+// TODO: rebuild event interface with Mitt.
+export class SuiWallet implements Wallet {
+    readonly #events: Emitter<WalletEventsMap>;
+    readonly #version = '1.0.0' as const;
+    readonly #name = 'Sui Wallet' as const;
+    #account: ReadonlyWalletAccount | null;
+    #messagesStream: WindowMessageStream;
+
+    get version() {
+        return this.#version;
+    }
+
+    get name() {
+        return this.#name;
+    }
+
+    get icon() {
+        // TODO: Improve this with ideally a vector logo.
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        return icon as any;
+    }
+
+    get chains() {
+        // TODO: Extract chain from wallet:
+        return SUI_CHAINS;
+    }
+
+    get features(): ConnectFeature & SuiSignAndExecuteTransactionFeature {
+        return {
+            'standard:connect': {
+                version: '1.0.0',
+                connect: this.#connect,
+            },
+            'sui:signAndExecuteTransaction': {
+                version: '1.0.0',
+                signAndExecuteTransaction: this.#signAndExecuteTransaction,
+            },
+        };
+    }
+
+    get accounts() {
+        return this.#account ? [this.#account] : [];
+    }
+
+    constructor() {
+        this.#events = mitt();
+        this.#account = null;
+        this.#messagesStream = new WindowMessageStream(
+            'sui_in-page',
+            'sui_content-script'
+        );
+
+        this.#connected();
+    }
+
+    on<E extends WalletEventNames>(
+        event: E,
+        listener: WalletEvents[E]
+    ): () => void {
+        this.#events.on('standard:change', listener);
+        return () => this.#events.off(event, listener);
+    }
+
+    #connected = async () => {
+        const accounts = await mapToPromise(
+            this.#send<GetAccount, GetAccountResponse>({
+                type: 'get-account',
+            }),
+            (response) => response.accounts
+        );
+
+        const properties: ('accounts' | 'chains')[] = [];
+
+        const [address] = accounts;
+
+        if (address) {
+            const account = this.#account;
+            if (!account || account.address !== address) {
+                this.#account = new ReadonlyWalletAccount({
+                    address,
+                    // TODO: Expose public key instead of address:
+                    publicKey: new Uint8Array(),
+                    chains: SUI_CHAINS,
+                    features: [
+                        'sui:signAndExecuteTransaction',
+                        'standard:signMessage',
+                    ],
+                });
+                properties.push('accounts');
+            }
+        }
+
+        if (properties.length) {
+            this.#events.emit('standard:change', properties);
+        }
+    };
+
+    #connect: ConnectMethod = async (input) => {
+        if (!input?.silent) {
```

Comment:
I don't understand why we need outside to pass `slient` to tell if the permission prompt should popup.
It should be judged by ext checking the permission approval history.
What use case would there be?

Reply:
The intent here is for auto-connecting to the users preferred wallet. You don’t want to prompt the user for connection, but if the permissions have already been granted then it can be connected to automatically.

Reply:
What about this case: if the `slient` passed in but no permission granted, wallet exts still need to prompt for approval

Reply:
The intent there is actually that the connect would just silently fail, without prompting the user.

Reply:
This could be a bug from user's perspective? Not convinced. More use cases?

Reply:
It couldn't be perceived be a bug because this is only for the auto-connect flow. The user will just see a button to connect their account as if it was not connected.

Reply:
Alright, then we follow your logic and try